lib/irb/context.rb

```diff
@@ -176,26 +176,19 @@ def initialize(irb, workspace = nil, input_method = nil)
       RegexpCompletor.new
     end
 
-    TYPE_COMPLETION_REQUIRED_PRISM_VERSION = '0.18.0'
-
     private def build_type_completor
-      unless Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.0') && RUBY_ENGINE != 'truffleruby'
-        warn 'TypeCompletion requires RUBY_VERSION >= 3.0.0'
+      if RUBY_ENGINE == 'truffleruby'
+        # Avoid SynatxError. truffleruby does not support endless method definition yet.
+        warn 'TypeCompletor is not supported on TruffleRuby yet'
         return
       end
+
       begin
-        require 'prism'
+        TypeCompletor.new(irb_path)
```

Comment:
Can we move `require 'repl_type_completor'` here? It will help readers to understand what the `LoadError` rescue is primarily for.